src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/HACK_NodeHelper.cs

```
@@ -0,0 +1,136 @@
+        }
+
+        /// <summary>
+        /// Select an item in a IVsIHierarchy.
+        /// </summary>
+        private static void Select(ConfiguredProject configuredProject, IServiceProvider serviceProvider, uint itemId)
+        {
+            UseWindow(configuredProject, serviceProvider, (hierarchy, window) =>
+            {
+                // We need to unselect the item if it is already selected to re-select it correctly.
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_UnSelectItem);
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_SelectItem);
+            });
+        }
+
+        /// <summary>
+        /// Expand a folder in a IVsUIHierarchy.
+        /// </summary>
+        private static void ExpandFolder(ConfiguredProject configuredProject, IServiceProvider serviceProvider, uint itemId)
+        {
+            UseWindow(configuredProject, serviceProvider, (hierarchy, window) =>
+            {
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_ExpandFolder);
+            });
+        }
+
+        /// <summary>
+        /// Collapse a folder in a IVsUIHierarchy.
+        /// </summary>
+        private static void CollapseFolder(ConfiguredProject configuredProject, IServiceProvider serviceProvider, uint itemId)
+        {
+            UseWindow(configuredProject, serviceProvider, (hierarchy, window) =>
+            {
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_CollapseFolder);
+            });
+        }
+
+        /// <summary>
+        /// Callbacks with a hierarchy and hierarchy window for use.
+        /// </summary>
+        private static void UseWindow(ConfiguredProject configuredProject, IServiceProvider serviceProvider, Action<IVsUIHierarchy, IVsUIHierarchyWindow> callback)
+        {
+            if (configuredProject.UnconfiguredProject.Services.HostObject is IVsUIHierarchy hierarchy)
```

Just do the cast, if this fails (which it won't) this will be easier to debug than silent failure.

---

Ok